slack/incident_commands.py

```diff
@@ -13,23 +13,31 @@ def send_help_text(incident: Incident, user_id: str, message: str):
 def update_summary(incident: Incident, user_id: str, message: str):
     incident.summary = message
     incident.save()
+
+    user = GetOrCreateSlackExternalUser(external_id=user_id)
+    Timeline.objects.new_event(incident=incident, author=user, source='slack_cmd_summary', text=message)
     return True, None
 
 
 @incident_command(['impact'], helptext='Explain the impact of this')
 def update_impact(incident: Incident, user_id: str, message: str):
     incident.impact = message
     incident.save()
+
+    user = GetOrCreateSlackExternalUser(external_id=user_id)
+    Timeline.objects.new_event(incident=incident, author=user, source='slack_cmd_impact', text=message)
     return True, None
 
 
 @incident_command(['lead'], helptext='Assign someone as the incident lead')
 def set_incident_lead(incident: Incident, user_id: str, message: str):
     assignee = reference_to_id(message) or user_id
-    name = get_user_profile(assignee)['name']
-    user = GetOrCreateSlackExternalUser(external_id=assignee, display_name=name)
+    user = GetOrCreateSlackExternalUser(external_id=assignee)
     incident.lead = user
     incident.save()
+
+    user = GetOrCreateSlackExternalUser(external_id=user_id)
```

Comment:
Should one of these `user` variable declarations be renamed for clarity or are we intentionally overriding the previous value from the assignee, to the user who triggered the action?

I think overriding might cause a bug on the line below for the timeline as it will say the lead has been set to the user who triggered the action (user_id), not who the assignee (assignee) has been set to if I understand correctly.

Reply:
P.s awesome to record this important events in the timeline 👍

Reply:
You are right. There was a bit of copy/paste and I did not really read the function. Should be good now. Thangs for all of your helpful reviews @AshleyPoole. They are appreciated.